packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -246,31 +263,35 @@ export class PipelineEditor extends React.Component<
       {
         action: 'export',
         label: 'Export Pipeline',
-        enable: true,
-        iconEnabled: Utils.getEncodedIcon(exportPipelineIcon, darkmode),
-        iconDisabled: Utils.getEncodedIcon(exportPipelineIcon, darkmode)
+        enable: !this.state.emptyPipeline,
+        iconEnabled: IconUtil.encode(exportPipelineIcon),
+        iconDisabled: IconUtil.encode(exportPipelineIcon)
       },
       {
         action: 'clear',
         label: 'Clear Pipeline',
-        enable: true,
-        iconEnabled: Utils.getEncodedIcon(clearPipelineIcon, darkmode),
-        iconDisabled: Utils.getEncodedIcon(clearPipelineIcon, darkmode)
+        enable: !this.state.emptyPipeline || !emptyCanvas,
+        iconEnabled: IconUtil.encode(clearPipelineIcon),
+        iconDisabled: IconUtil.encode(clearPipelineIcon)
       },
       { divider: true },
-      { action: 'undo', label: 'Undo', enable: true },
-      { action: 'redo', label: 'Redo', enable: true },
-      { action: 'cut', label: 'Cut', enable: false },
-      { action: 'copy', label: 'Copy', enable: false },
-      { action: 'paste', label: 'Paste', enable: false },
+      { action: 'undo', label: 'Undo' },
+      { action: 'redo', label: 'Redo' },
+      { action: 'cut', label: 'Cut' },
+      { action: 'copy', label: 'Copy' },
+      { action: 'paste', label: 'Paste' },
       { action: 'addComment', label: 'Add Comment', enable: true },
-      { action: 'delete', label: 'Delete', enable: true },
+      { action: 'delete', label: 'Delete' },
       {
         action: 'arrangeHorizontally',
         label: 'Arrange Horizontally',
-        enable: true
+        enable: !this.state.emptyPipeline || !emptyCanvas
       },
-      { action: 'arrangeVertically', label: 'Arrange Vertically', enable: true }
+      {
+        action: 'arrangeVertically',
+        label: 'Arrange Vertically',
+        enable: !this.state.emptyPipeline || !emptyCanvas
```

Comment:
upon further testing this is needed but should prob just be `!this.state.emptyPipeline` since comments are not included when the arrange buttons are used, so leaving it enabled when there's only comments does not make sense